### DISCUSSION TITLE
Eluna script error：attempt to index global 'spell' (a nil value)

### DISCUSSION THREAD

**Discussion by felixjiang** [azerothcore/azerothcore-wotlk#11846] [General]
I wrote a logic (code below) to use Fire Ball (42833) to trigger the removal of Harvest Soul (28679) that Gothik the Harvester from Naxxramas casts on me.

When worldserver.exe is loaded, the command window prints an error: attempt to index global 'spell' (a nil value).

Although the code still works, but I hate to see errors. Any idea how to mitigate this error?

```
local function RemoveHarvestSoul(event, player)
    spell = player:GetCurrentSpell(1)
    if ((spell:GetEntry() == 42833) and player:HasAura(28679))
    then
        player:RemoveAura(28679)
    end
end
 
RegisterPlayerEvent(5, RemoveHarvestSoul)
```

**Comment by Shoro2** [comment]
`PLAYER_EVENT_ON_SPELL_CAST              =     5,        // (event, player, spell, skipCheck)`

the spell is already given with the event, no need to `spell = player:GetCurrentSpell(1)`

so changing it to:
```
local function RemoveHarvestSoul(event, player, spell, skipCheck)
    if ((spell:GetEntry() == 42833) and player:HasAura(28679))
    then
        player:RemoveAura(28679)
    end
end
 
RegisterPlayerEvent(5, RemoveHarvestSoul)
```
should make it work without any errors
